README.md

```diff
@@ -170,54 +168,75 @@ sensor:
+      - { phase_id: phase_b, input:  "8", power: { id:  cir8, filters: [ *pos ] } }
+      - { phase_id: phase_b, input:  "9", power: { id:  cir9, filters: [ *pos ] } }
+      - { phase_id: phase_b, input: "10", power: { id: cir10, filters: [ *pos ] } }
+      - { phase_id: phase_a, input: "11", power: { id: cir11, filters: [ *pos, multiply: 2 ] } }
+      - { phase_id: phase_a, input: "12", power: { id: cir12, filters: [ *pos, multiply: 2 ] } }
+      - { phase_id: phase_a, input: "13", power: { id: cir13, filters: [ *pos ] } }
+      - { phase_id: phase_a, input: "14", power: { id: cir14, filters: [ *pos ] } }
+      - { phase_id: phase_b, input: "15", power: { id: cir15, filters: [ *pos ] } }
+      - { phase_id: phase_a, input: "16", power: { id: cir16, filters: [ *pos ] } }
+    on_update:
+      then:
+        - component.update: total_power
+        - component.update: balance_power
+  - { platform: copy, name: "Phase A Power", source_id: phase_a_power, filters: *throttle_avg }
+  - { platform: copy, name: "Phase B Power", source_id: phase_b_power, filters: *throttle_avg }
+  - { platform: copy, name: "Total Power", source_id: total_power, filters: *throttle_avg }
+  - { platform: copy, name: "Balance Power", source_id: balance_power, filters: *throttle_avg }
+  - { platform: copy, name:  "Circuit 1 Power", source_id:  cir1, filters: *throttle_avg }
+  - { platform: copy, name:  "Circuit 2 Power", source_id:  cir2, filters: *throttle_avg }
+  - { platform: copy, name:  "Circuit 3 Power", source_id:  cir3, filters: *throttle_avg }
+  - { platform: copy, name:  "Circuit 4 Power", source_id:  cir4, filters: *throttle_avg }
+  - { platform: copy, name:  "Circuit 5 Power", source_id:  cir5, filters: *throttle_avg }
+  - { platform: copy, name:  "Circuit 6 Power", source_id:  cir6, filters: *throttle_avg }
+  - { platform: copy, name:  "Circuit 7 Power", source_id:  cir7, filters: *throttle_avg }
+  - { platform: copy, name:  "Circuit 8 Power", source_id:  cir8, filters: *throttle_avg }
+  - { platform: copy, name:  "Circuit 9 Power", source_id:  cir9, filters: *throttle_avg }
+  - { platform: copy, name: "Circuit 10 Power", source_id: cir10, filters: *throttle_avg }
+  - { platform: copy, name: "Circuit 11 Power", source_id: cir11, filters: *throttle_avg }
+  - { platform: copy, name: "Circuit 12 Power", source_id: cir12, filters: *throttle_avg }
+  - { platform: copy, name: "Circuit 13 Power", source_id: cir13, filters: *throttle_avg }
+  - { platform: copy, name: "Circuit 14 Power", source_id: cir14, filters: *throttle_avg }
+  - { platform: copy, name: "Circuit 15 Power", source_id: cir15, filters: *throttle_avg }
+  - { platform: copy, name: "Circuit 16 Power", source_id: cir16, filters: *throttle_avg }
   - platform: template
-    name: "Total Power"
     lambda: return id(phase_a_power).state + id(phase_b_power).state;
-    update_interval: 2.88s
+    update_interval: never
```

Comment:
I think that `update_interval: 0s` is what you're looking for here? That will automatically run the loop often enough to poll the hardware at a good rate. When a new value comes in, `on_value` will be called.

Reply:
The ``on_update`` trigger will force these template components to update once all the power sensors have updated after a measurement. We do not need it to update more than that, so doing it every loop would be excessive. We could also run into floating point precision issues with the daily energy component if we ask it to update very frequently with very small intervals.

Reply:
I just added a comment to each of these lines now to make this hopefully more clear, because at first glance it is not obvious!